In [2]:
from pulp import LpMaximize, LpProblem, LpVariable, lpSum, value

# Step 1: Define the problem
model = LpProblem(name="profit-maximization", sense=LpMaximize)

# Step 2: Define decision variables
x = LpVariable(name="Product_A", lowBound=0, cat='Continuous')
y = LpVariable(name="Product_B", lowBound=0, cat='Continuous')

# Step 3: Define objective function (maximize profit)
model += 20 * x + 30 * y, "Total Profit"

# Step 4: Add constraints
model += 2 * x + 4 * y <= 100, "Labor Constraint"
model += 3 * x + 2 * y <= 90, "Material Constraint"

# Step 5: Solve the problem
model.solve()

# Step 6: Output the results
print(f"Produce {x.varValue:.0f} units of Product A")
print(f"Produce {y.varValue:.0f} units of Product B")
print(f"Maximum Profit = ₹{value(model.objective)}")


Produce 20 units of Product A
Produce 15 units of Product B
Maximum Profit = ₹850.0


### Problem Statement:
Maximize profit by determining how many units of Product A and B to produce under labor and material constraints.

### Approach:
Used Linear Programming via PuLP to model constraints and objective. Solved using `model.solve()`.

### Insights:
- Best to produce 20 units of A and 15 units of B
- Utilizes all resources efficiently
- Achieves a maximum profit of ₹850
